vmdb/app/models/ems_refresh/parsers/foreman.rb

```diff
@@ -0,0 +1,88 @@
+module EmsRefresh::Parsers
+  class Foreman
+    def self.ems_inv_to_hashes(inv)
+      new.ems_inv_to_hashes(inv)
+    end
+
+    def ems_inv_to_hashes(inv)
+      result = {}
+      ids = {}
+      result[:customization_scripts] = customization_scripts_inv_to_hash(inv[:media], inv[:ptables])
+      add_ids(ids, result[:customization_scripts])
+      result[:operating_system_flavors] = operating_system_flavors_inv_to_hashes(inv[:operating_systems], ids)
+      add_ids(ids, result[:operating_system_flavors])
+      result[:configuration_profiles] = configuration_profile_inv_to_hashes(inv[:hostgroups], ids)
+      add_ids(ids, result[:configuration_profiles])
+      result[:configured_systems] = configured_system_inv_to_hashes(inv[:hosts], ids)
+      result
+    end
+
+    def media_inv_to_hashes(media)
+      media.map do |m|
+        {
+          "manager_ref"  => "medium:#{m["id"]}",
```

Comment:
1. I don't like the current implementation.
2. I did see `name:#{id}` used in another provider
3. It is going to be tricky to support this in general. agree that it is best to not modify fields
4. foreman is not consistent with their naming. so you'll notice `operatingsystem_id` vs `operating_system`, making it even worse.
5. I don't like the multi key ids, but I suppose they may be necessary.